### DISCUSSION TITLE
An API platform?

### DISCUSSION THREAD

**Discussion by andrewm4894** [LAION-AI/Open-Assistant#3350] [Q&A]
Wondering if any plans for a sort of API platform or is there already some way to use OA via api's and completion endpoints etc?

Or is it mainly just via the UI for now?

**Comment by CoolOppo** [comment]
Ditto

**Comment by andreaskoepf** [comment]
While I fully can understand the wish for a API access (which would technically no problem, since we use our API internally) it will be very challenging to offer a free API from our side - at least not without some kind of a refinancing system. Please see https://github.com/LAION-AI/Open-Assistant/issues/2806

**Comment by andrewm4894** [comment]
I'd be happy to pay for such a thing and think that would be totally fine. Being able to interact with the UI for free (on some best effort basis) is one thing, but I defo would not be expecting to be able to use API access for free. So maybe could be something worth exploring.

**Comment by scottleedavis** [comment]
The chat believes it already exists...
![Screen Shot 2023-04-23 at 11 19 44 AM](https://user-images.githubusercontent.com/2914443/233857883-e5bf2460-bc44-4345-99c0-d45bcbb0ffb5.png)

**Comment by scottleedavis** [comment]
without this... integration is hackish and limited.  I do ai guests in a podcast for example...  which has code that relies on something like as follows:

```
# Chat with ChatGPT-4 function
def chat_with_chatgpt(text):
    response = openai.ChatCompletion.create(
        model="gpt-3.5-turbo",
        messages=[
            {"role": "user", "content": text}
        ]
    )
    return response.choices[0].message.content.strip()
```
